src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -192,29 +192,12 @@ install_packages_in_root (RpmOstreeTreeComposeContext  *self,
                                             NULL);
   gs_free char *ret_new_inputhash = NULL;
 
-  /* We can be control-c'd at any time */
-#if BUILDOPT_HAVE_RPMSQ_SET_INTERRUPT_SAFETY
-  rpmsqSetInterruptSafety (FALSE);
-#endif
-
-  /* Apparently there's only one process-global macro context;
-   * realistically, we're going to have to refactor all of the RPM
-   * stuff to a subprocess.
-   */
-  hifctx = hif_context_new ();
-  hif_context_set_http_proxy (hifctx, opt_proxy ? opt_proxy : g_getenv ("http_proxy"));
-
+  hifctx = _rpmostree_libhif_get_default ();
```

Comment:
Need to call `hif_context_set_http_proxy()` here if `opt_proxy` is set, or else `--proxy` breaks.